src/lib/netlist/plib/ppmf.h

```diff
@@ -153,277 +119,341 @@ namespace plib {
+				func = *reinterpret_cast<generic_function *>(vtable_base + m_function + m_this_delta - 1);
 			}
 		}
 
 		// actual state
 		uintptr_t               m_function;         // first item can be one of two things:
 													//    if even, it's a pointer to the function
 													//    if odd, it's the byte offset into the vtable
-		int                     m_this_delta;       // delta to apply to the 'this' pointer
+		ptrdiff_t               m_this_delta;       // delta to apply to the 'this' pointer
+	};
+
+	template <>
+	class mfp_raw<PPMF_TYPE_INTERNAL_MSC>
+	{
+	public:
+		// construct from any member function pointer
+		using generic_function = void (*)();
+		struct unknown_base_equiv { generic_function fptr; int thisdisp, vptrdisp, vtdisp; };
+		struct single_base_equiv { generic_function fptr; };
+
+		template<typename MemberFunctionType>
+		mfp_raw(MemberFunctionType mftp)
+		: m_function(0), m_this_delta(0), m_vptr_offs(0), m_vt_index(0), m_size(0)
+		{
+			static_assert(sizeof(*this) >= sizeof(MemberFunctionType), "size mismatch");
+			*reinterpret_cast<MemberFunctionType *>(this) = mftp; // NOLINT
+			m_size = sizeof(mftp); //NOLINT
+		}
+
+	//private:
+		// extract the generic function and adjust the object pointer
+		void convert_to_generic(generic_function &func, mfp_generic_class *&object) const
+		{
+			// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+			auto *byteptr = reinterpret_cast<std::uint8_t *>(object);
+
+			// test for pointer to member function cast across virtual inheritance relationship
+			if ((sizeof(unknown_base_equiv) == m_size) && m_vt_index)
+			{
+				// add offset from "this" pointer to location of vptr, and add offset to virtual base from vtable
+				byteptr += m_vptr_offs;
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				std::uint8_t const *const vptr = *reinterpret_cast<std::uint8_t const *const *>(byteptr);
+				// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+				byteptr += *reinterpret_cast<int const *>(vptr + m_vt_index);
+			}
+
+			// add "this" pointer displacement if present in the pointer to member function
+			if (sizeof(single_base_equiv) < m_size)
+				byteptr += m_this_delta;
+			// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+			object = reinterpret_cast<mfp_generic_class *>(byteptr);
+
+			// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast,performance-no-int-to-ptr)
+			auto const *funcx = reinterpret_cast<std::uint8_t const *>(m_function);
+			// NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast,performance-no-int-to-ptr)
+			func = reinterpret_cast<generic_function>(std::uintptr_t(funcx));
+		}
+
+		// actual state
+		uintptr_t m_function;         // first item can be one of two things:
+													//    if even, it's a pointer to the function
+													//    if odd, it's the byte offset into the vtable
+		int       m_this_delta;       // delta to apply to the 'this' pointer
```

Comment:
The copy/pasted comment is here, too – the first item is always a function pointer with the MSVC ABI.